extensions/amp-experiment/1.0/attribute-allow-list/allowed-attribute-mutation-entry.js

```diff
@@ -93,3 +93,22 @@ export class DefaultAllowedURLAttributeMutationEntry extends AllowedAttributeMut
     return true;
   }
 }
+
+/**
+ * Class for handling class type attribute
+ * mutations. E.g ['class']['*']
+ */
+export class DefaultClassAllowedAttributeMutationEntry extends AllowedAttributeMutationEntry {
+  /** @override */
+  validate(mutationRecord) {
+    const value = mutationRecord['value'];
+
+    // Don't allow the .i-amphtml class
+    // See `validator/validator-main.protoascii`
+    if (value.match(/(^|\\W)i-amphtml-/)) {
```

Comment:
Can we declare a constant for this instead of inlining the regex

Reply:
Talked offline, since we are having a refactor, and the current code is like this. We went ahead and added some more comments, and will address this in the refactor 😄